misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/connector/generator/ConnectorGenerator.java

```diff
@@ -0,0 +1,290 @@
+                        defaultableParameter.paramName().get().text() : "";
+                param.displayAnnotation = getDisplayAnnotationFromAnnotationsList(
+                        defaultableParameter.annotations());
+                param.defaultValue = defaultableParameter.expression().toString();
+                param.documentation = getParameterDocFromMetadataList(param.name, optionalMetadataNode);
+                parameters.add(param);
+            } else if (parameterNode instanceof IncludedRecordParameterNode) {
+                IncludedRecordParameterNode includedRecord = (IncludedRecordParameterNode) parameterNode;
+                InclusionType param = new InclusionType(Type.fromSyntaxNode(includedRecord.typeName(), semanticModel));
+                param.name = includedRecord.paramName().isPresent() ? includedRecord.paramName().get().text() : "";
+                param.displayAnnotation = getDisplayAnnotationFromAnnotationsList(includedRecord.annotations());
+                param.documentation = getParameterDocFromMetadataList(param.name, optionalMetadataNode);
+                parameters.add(param);
+            }
+        }
+
+        return parameters;
+    }
+
+    public static Type getReturnParameter(ReturnTypeDescriptorNode returnTypeDescriptorNode,
+                                          Optional<MetadataNode> optionalMetadataNode,
+                                          SemanticModel semanticModel) {
+        Type parameter = Type.fromSyntaxNode(returnTypeDescriptorNode.type(), semanticModel);
+        parameter.displayAnnotation = getDisplayAnnotationFromAnnotationsList(returnTypeDescriptorNode.annotations());
+        parameter.documentation = getParameterDocFromMetadataList("return", optionalMetadataNode);
+        return parameter;
+    }
+
+    public static Map<String, String> getDisplayAnnotationFromAnnotationsList(NodeList<AnnotationNode> annotations) {
+        Map<String, String> displayAnnotation = new HashMap<>();
+        Optional<AnnotationNode> optDisplayAnnotation = annotations.stream()
+                .filter(annotationNode -> annotationNode.annotReference().toString().equals("display ")).findAny();
+        if (optDisplayAnnotation.isPresent()) {
+            if (optDisplayAnnotation.get().annotValue().isPresent()) {
+                Map<String, String> finalDisplayAnnotation = displayAnnotation;
+                optDisplayAnnotation.get().annotValue().get().fields().forEach(mappingFieldNode -> {
+                    if (mappingFieldNode instanceof SpecificFieldNode) {
+                        SpecificFieldNode fieldNode = (SpecificFieldNode) mappingFieldNode;
+                        finalDisplayAnnotation.put(fieldNode.fieldName().toString(),
+                                fieldNode.valueExpr().isPresent() ? fieldNode.valueExpr().get()
+                                        .toSourceCode().replace("\"", "") : "");
```

Comment:
Can we do something like this ?

```suggestion
if (fieldNode.fieldName().toString().equals("label") || fieldNode.fieldName().toString().equals("iconPath") ) {
    finalDisplayAnnotation.put(fieldNode.fieldName().toString(), fieldNode.valueExpr().isPresent() ? fieldNode.valueExpr().get().toSourceCode().replace("\"", "") : "");
} else {
    finalDisplayAnnotation.put(fieldNode.fieldName().toString(), fieldNode.valueExpr().isPresent() ? fieldNode.valueExpr().get().toSourceCode() : "");
}
```

What do you think ?

Reply:
sure, I will update like this